config/env.config.js

```diff
@@ -85,15 +91,38 @@ class RedirectPlugin {
                                     return match[1];
                                 }
                                 return null;
-                            };
+                            }
+
+                            function isAllowedUrl(url) {
+                                if (url == null || url.trim().length === 0) {
+                                    return true;
+                                }
+                                
+                                url = decodeURIComponent(url);
+
+                                var validRedirectLocations = ${JSON.stringify(validRedirectLocations)};
+
+                                var matchingLocation = validRedirectLocations.find(function(location) {
+                                    return location.indexOf(url) === 0;
```

Comment:
We should probably do an exact match, not just looking for a substring as someone could register a new domain.

Reply:
We're effectively looking at starts-with (indexOf === 0).  I think that's better, because we don't want query parameters or etc to mess up the logic.  As long as we start with a fully-qualified domain that's in our whitelist, we're ok.